SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -0,0 +1,292 @@
+PATH=\"\$VIRTUAL_ENV/bin:\$PATH\"
+export PATH
+
+# unset PYTHONHOME if set
+# this will fail if PYTHONHOME is set to the empty string (which is bad anyway)
+# could use \`if (set -u; : \$PYTHONHOME) ;\` in bash
+if [ -n \"\${PYTHONHOME:-}\" ] ; then
+    _OLD_VIRTUAL_PYTHONHOME=\"\${PYTHONHOME:-}\"
+    unset PYTHONHOME
+fi
+if [ -n \"\${PYTHONPATH:-}\" ] ; then # added line
+    _OLD_VIRTUAL_PYTHONPATH=\"\${PYTHONPATH:-}\" # added line
+    unset PYTHONPATH # added line
+fi # added line
+
+if [ -z \"\${VIRTUAL_ENV_DISABLE_PROMPT:-}\" ] ; then
+    _OLD_VIRTUAL_PS1=\"\${PS1:-}\"
+    PS1=\"(${ENV_NAME}) \${PS1:-}\"
+    export PS1
+fi
+
+# This should detect bash and zsh, which have a hash command that must
+# be called to get it to forget past commands.  Without forgetting
+# past commands the \$PATH changes we made may not be respected
+if [ -n \"\${BASH:-}\" -o -n \"\${ZSH_VERSION:-}\" ] ; then
+    hash -r 2> /dev/null
+fi
+"
+
+# Create a new bash session to avoid conflicts with the current environment in the background
+env -i bash --noprofile --norc << EOF
+
+export OAUTH2_FILE=${OAUTH2_FILE}
+export OAUTH2_TOKEN=${OAUTH2_TOKEN}
+export KRB5CCNAME=${KRB5CCNAME}
+export KRB5CCNAME_NB_TERM=${KRB5CCNAME_NB_TERM}
+
+if [ -n "$ACCPY_CUSTOM_VERSION" ]; then
+    source ${ACCPY_BASE}/${ACCPY_CUSTOM_VERSION}/setup.sh
+    if [ -d "${ENV_PATH}" ] && [ -n "${CLEAR_ENV}" ]; then
+        rm -rf ${ENV_PATH}
+    fi
+    acc-py venv ${ENV_PATH}
+else
+    if [ -d "${ENV_PATH}" ]; then
+        echo "Recreating (--clear) virtual environment ${ENV_NAME} using Python (${PYTHON_PATH})..."
+    else
+        echo "Creating virtual environment ${ENV_NAME} using Python (${PYTHON_PATH})..."
+    fi
+    ${PYTHON_PATH} -m venv ${ENV_PATH} ${CLEAR_ENV} --copies
+fi
+
+mkdir -p /home/$USER/.local/share/jupyter/kernels
+ln -f -s ${ENV_PATH}/share/jupyter/kernels/${ENV_NAME} /home/$USER/.local/share/jupyter/kernels/${ENV_NAME}
+
+echo "Setting up the virtual environment..."
+source ${ENV_PATH}/bin/activate
+
+# Check if ACCPY_CUSTOM_VERSION is not set and ipykernel is on the requirements file, if not, add the latest version
+if [ -z "$ACCPY_CUSTOM_VERSION" ] && [ -z "$(grep -i 'ipykernel' ${REQ_PATH})" ]; then
+    echo -e "\nipykernel" >> ${REQ_PATH}
+fi
+
+echo "Installing packages from ${REQ_PATH}..."
+pip install -r ${REQ_PATH}
+
+python -m ipykernel install --name ${ENV_NAME} --display-name "Python (${ENV_NAME})" --prefix ${ENV_PATH}
+
+# Remove ipykernel package from the requirements file, if it was added
+if [ -z "$ACCPY_CUSTOM_VERSION" ] && [ -z "$(grep -i 'ipykernel' ${REQ_PATH})" ]; then
+    sed -i '/\nipykernel/d' ${REQ_PATH}
+fi
+
+# Copy the requirements file to the virtual environment
+cp ${REQ_PATH} ${ENV_PATH}
+
+# Get (de)activate script to be independent from the PYTHONPATH, only if ACCPY_CUSTOM_VERSION is not set
+if [ -z "$ACCPY_CUSTOM_VERSION" ]; then
+    echo '${ACTIVATE_BIN_TEMPLATE}' > ${ENV_PATH}/bin/activate
+fi
```

Comment:
This would go away if my assumption that we do not need a custom deactivate anymore is true.